apm-sniffer/apm-sdk-plugin/light4j-plugins/light4j-plugin/src/main/java/org/apache/skywalking/apm/plugin/light4j/HandleRequestInterceptor.java

```diff
@@ -0,0 +1,141 @@
+import org.apache.skywalking.apm.agent.core.context.CarrierItem;
+import org.apache.skywalking.apm.agent.core.context.ContextCarrier;
+import org.apache.skywalking.apm.agent.core.context.ContextManager;
+import org.apache.skywalking.apm.agent.core.context.ContextSnapshot;
+import org.apache.skywalking.apm.agent.core.context.tag.Tags;
+import org.apache.skywalking.apm.agent.core.context.trace.AbstractSpan;
+import org.apache.skywalking.apm.agent.core.context.trace.SpanLayer;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.EnhancedInstance;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.InstanceMethodsAroundInterceptor;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.MethodInterceptResult;
+import org.apache.skywalking.apm.network.trace.component.ComponentsDefine;
+
+import java.lang.reflect.Method;
+
+/**
+ * {@link HandleRequestInterceptor} creates an entry span before the execution of
+ * {@link com.networknt.exception.ExceptionHandler#handleRequest(HttpServerExchange)} in the I/O thread.
+ *
+ * If the {@link Config.Plugin.Light4J#TRACE_HANDLER_CHAIN} flag is set, additionally a local span is produced for each
+ * {@link com.networknt.handler.MiddlewareHandler} and business handler before their respective
+ * {@link com.networknt.handler.LightHttpHandler#handleRequest(HttpServerExchange)} method executes.
+ * Since {@link com.networknt.handler.LightHttpHandler} is implemented by various middleware and business handlers and
+ * the Light4J framework delegates to these in succession, a chain of
+ * {@link org.apache.skywalking.apm.agent.core.context.trace.LocalSpan}s will be produced.
+ *
+ * @author tsuilouis
+ */
+public class HandleRequestInterceptor implements InstanceMethodsAroundInterceptor {
+
+    @Override
+    public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes,
+                             MethodInterceptResult result) {
+        if (isExceptionHandler(objInst)) {
+            HttpServerExchange exchange = (HttpServerExchange) allArguments[0];
+
+            if (exchange.isInIoThread()) {
+
+                String operationName = exchange.getRequestPath() + "@" + exchange.getRequestMethod();
+                final HeaderMap headers = exchange.getRequestHeaders();
+                final ContextCarrier contextCarrier = new ContextCarrier();
+
+                CarrierItem next = contextCarrier.items();
+                while (next.hasNext()) {
+                    next = next.next();
+                    next.setHeadValue(headers.getFirst(next.getHeadKey()));
+                }
+
+                AbstractSpan span = ContextManager.createEntrySpan(operationName, contextCarrier);
+                Tags.URL.set(span, exchange.getRequestURL());
+                Tags.HTTP.METHOD.set(span, exchange.getRequestMethod().toString());
+                span.setComponent(ComponentsDefine.LIGHT_4J);
```

Comment:
As you have light4j entry span, I think you should have a PR to submit icon showing in topology, in this repo https://github.com/apache/skywalking-rocketbot-ui

Reply:
Should be an icon shows up at the top of the node in the topology to represent this service providing service through Light4j.

Reply:
Ok, I have submitted the PR at https://github.com/apache/skywalking-rocketbot-ui/pull/157.

Reply:
Merged, you need to include the UI submodule update into this PR too.

Reply:
Ok, I have added a commit with the submodule update at https://github.com/apache/skywalking/pull/3323/commits/efde0fff50ddf44bb78f3e25b6d0ccd6f7c38d2f.